pkg/logging/fields.go

```diff
@@ -32,90 +39,123 @@ var resourceTypeFields = map[string]string{
+		context.add(ClusterID(clusterID[0]))
+	}
+	return zap.Inline(context)
+}
+
 // Err wraps err into a zap.Field instance with a well-known name 'error'.
 func Err(err error) zap.Field {
 	return zap.Error(err)
 }
 
 // ImageName provides the image name as a structured log field.
 func ImageName(name string) zap.Field {
-	return zap.String(imageField, name)
+	return String(imageField, name)
 }
 
 // ClusterID provides the cluster ID as a structured log field.
 func ClusterID(id string) zap.Field {
-	return zap.String(clusterIDField, id)
+	return String(clusterIDField, id)
 }
 
 // ImageID provides the image ID as a structured log field.
 func ImageID(id string) zap.Field {
-	return zap.String(imageIDField, id)
+	return String(imageIDField, id)
 }
 
 // NodeID provides the node ID as a structured log field.
 func NodeID(id string) zap.Field {
-	return zap.String(nodeIDField, id)
+	return String(nodeIDField, id)
 }
 
 // BackupName provides the backup name as a structured log field.
 func BackupName(name string) zap.Field {
-	return zap.String(backupField, name)
+	return String(backupField, name)
 }
 
 // CloudSourceName provides the cloud source name as a structured log field.
 func CloudSourceName(name string) zap.Field {
-	return zap.String(cloudSourceField, name)
+	return String(cloudSourceField, name)
 }
 
 // NotifierName provides the notifier name as a structured log field.
 func NotifierName(name string) zap.Field {
-	return zap.String(notifierField, name)
+	return String(notifierField, name)
 }
 
 // ErrCode refers to a specific human-readable error code. The error code
 // will be used to identify a specific issue and may be used to render e.g. help information
 // or can be used for filtering.
 func ErrCode(code string) zap.Field {
-	return zap.String(errCodeField, code)
+	return String(errCodeField, code)
 }
 
 // AlertID provides the alert ID as a structured log field.
 func AlertID(id string) zap.Field {
-	return zap.String(alertIDField, id)
+	return String(alertIDField, id)
 }
 
 // APITokenID provides the API token ID as a structured log field.
 func APITokenID(id string) zap.Field {
-	return zap.String(apiTokenIDField, id)
+	return String(apiTokenIDField, id)
 }
 
 // APITokenName provides the API token name as a structured log field.
 func APITokenName(name string) zap.Field {
-	return zap.String(apiTokenNameField, name)
+	return String(apiTokenNameField, name)
 }
 
 // ClusterName provides the cluster name as a structured log field.
 func ClusterName(name string) zap.Field {
-	return zap.String(clusterNameField, name)
+	return String(clusterNameField, name)
 }
 
 // Wrapper functions for zap.Field functions.
 
 // String provides a wrapper around zap.String and adds the key-value pair as structured log field.
 // This should be _always_ preferred over direct calls to zap to minimize dependency to it.
 func String(field, value string) zap.Field {
+	if value == "" {
```

Comment:
Currently we log empty values such as `request_id: ""`. I think this unnecessarily bloats the logs, so I changed the behavior to skip the field when empty.

Reply:
I think it's easier to check this in the logging package. Compare:

```
log.Errorw("Failed to acquire scan semaphore",
  logging.FromContext(ctx),
  logging.ImageName(request.GetImage().GetName().GetFullName()),
  logging.ImageID(request.GetImage().GetId()),
  logging.Err(err),
)
```
vs

```
fields := []zap.Field{logging.FromContext(ctx), logging.Err(err)}
if img := request.GetImage().GetName().GetFullName(); img != "" {
  fields = append(fields, logging.ImageName(img))
}
if id := request.GetImage().GetName().GetId(); id != "" {
  fields = append(fields, logging.ImageId(id))
}
log.Errorw("Failed to acquire scan semaphore", fields)
```

You can see how checking on the caller site can quickly get out of hand by introducing a bunch of conditional code.